### PR TITLE
MIL-2258 stock stories language filter

### PR DESCRIPTION
**Pull request by dwatring**

### CODE REVIEW

File: src/apis/marketData.ts [hunk 1]
```diff
@@ -249,6 +249,10 @@ export interface MarketUpcomingEarningsResponse {
     reportDate: string
 }
 
+export interface MarketUpcomingEarningsCacheResponse extends MarketUpcomingEarningsResponse {
+    marketcap: number
```
Comment (dmoss18) [inline src/apis/marketData.ts]: this changed to `marketCap`

Comment (dmoss18) [reply src/apis/marketData.ts]: Also it should have `symbol` and `reportDate` in it

File: src/client.ts [hunk 2]
```diff
@@ -421,10 +422,20 @@ export default class IEXClient {
...
    */
-  public upcomingEarnings(): Promise<MarketDataAPI.MarketUpcomingEarningsResponse[]> {
-    return this.request('/stock/market/upcoming-earnings')
+  public upcomingEarnings(
+    params: {
+      date?: string,
+      year?: number,
+      month?: number,
+      week?: number,
+      startDate?: string,
```
Comment (dmoss18) [inline src/client.ts]: Can you change the type to `Date` for `startDate`, `endDate`, and `date` and then just format it into a string in this method?

File: src/apis/marketData.ts (hunk 1 above)
Comment (dwatring) [reply src/apis/marketData.ts]: its extending `MarketUpcomingEarningsResponse`
``` {
    symbol: string
    reportDate: string
}```

Comment (dmoss18) [reply src/apis/marketData.ts]: ah cool

**Pull request merged**